### ISSUE TITLE
Nested field is displayed with its elements in arbitrary order, e.g. SUM(VALUE(position))

### ISSUE THREAD

**Issue by blais** [beancount/beanquery#40]
Original report by Adam Wolenc [adamdblu](https://bitbucket.org/adamdblu).

When displaying a nested target, for example a `SUM` of `position`s containing more than one currency, bean-query displays the elements of this field in arbitrary order. Is there a way to tell it to sort them by currency?

This is best illustrated by an example.
```
2000-01-04 open Assets:HOOL:Unvested:FUTURE2000 HOOL.UNVEST
2000-01-04 open Income:HOOL:Awards              HOOL.UNVEST
2001-01-04 open Assets:HOOL:RSURefund           USD
2001-01-04 open Income:HOOL:HoolStockUnit       USD

2000-01-04 * "Award FUTURE2000"
  Income:HOOL:Awards                    -10 HOOL.UNVEST
  Assets:HOOL:Unvested:FUTURE2000        10 HOOL.UNVEST

2001-01-04 * "Vest FUTURE2000"
  Income:HOOL:HoolStockUnit        -1000.00 USD
  Assets:HOOL:RSURefund
```

Sub accounts under Assets:HOOL now contain two currencies.

Running this query repeatedly results in one of two outputs arbitrarily.

```sql
SELECT ROOT(account, 2) AS institution,
       SUM(VALUE(position)) AS mkt
 GROUP BY institution
 ORDER BY mkt, institution DESC;

```

```
institution         mkt         
----------- --------------------
Assets:HOOL    10    HOOL.UNVEST,  1000.00 USD        
Income:HOOL   -10    HOOL.UNVEST, -1000.00 USD   

```

```
institution         mkt         
----------- --------------------
Assets:HOOL  1000.00 USD        ,    10    HOOL.UNVEST
Income:HOOL -1000.00 USD        ,   -10    HOOL.UNVEST
```

**Comment by dnicolodi** [comment]
I believe the issue was due to the fact that Inventories are dictionaries and their string representation is obtained iterating the items in the dictionary. When Python did not have dictionary with deterministic order this resulted in the elements being displayed in different order. The main issue went away since Python adopted deterministic order dictionary: Inventories are not displayed with the different entries always in the same stable but arbitrary order.

To make the order predictable I changed the code to sort the elements before computing the string representation. 

Inventory display was broken in other ways. I modified is significantly in #87
